design-documents/api-clean-up/2.5/api-candidates.md

```diff
@@ -0,0 +1,1522 @@
+
+# Factories for API classes/interfaces
+
+1. app/code/Magento/Catalog/Block/Product/ImageFactory.php
+1. app/code/Magento/Catalog/Model/Product/Type/Price/Factory.php
+1. app/code/Magento/Config/Block/System/Config/Form/Field/Factory.php
+1. app/code/Magento/Config/Block/System/Config/Form/Fieldset/Factory.php
+1. app/code/Magento/Customer/Model/Metadata/FormFactory.php
+1. app/code/Magento/ImportExport/Model/Export/Factory.php
+1. app/code/Magento/Quote/Model/Quote/Address/RateCollectorInterfaceFactory.php
+1. app/code/Magento/Quote/Model/Quote/Address/TotalFactory.php
+1. app/code/Magento/Sales/Model/Order/CreditmemoFactory.php
+1. app/code/Magento/Sales/Model/ResourceModel/Order/Collection/Factory.php
+1. app/code/Magento/Ui/Component/Form/Fieldset/Factory.php
+1. lib/internal/Magento/Framework/Acl/AclResourceFactory.php
+1. lib/internal/Magento/Framework/AclFactory.php
+1. lib/internal/Magento/Framework/Amqp/ConfigFactory.php
+1. lib/internal/Magento/Framework/Api/ExtensionAttribute/JoinDataInterfaceFactory.php
+1. lib/internal/Magento/Framework/Api/Search/DocumentFactory.php
+1. lib/internal/Magento/Framework/Api/Search/SearchCriteriaFactory.php
+1. lib/internal/Magento/Framework/App/Config/ValueFactory.php
+1. lib/internal/Magento/Framework/Config/Data/ConfigDataFactory.php
+1. lib/internal/Magento/Framework/Config/ViewFactory.php
+1. lib/internal/Magento/Framework/Controller/Result/JsonFactory.php
+1. lib/internal/Magento/Framework/DB/Adapter/Pdo/MysqlFactory.php
+1. lib/internal/Magento/Framework/DB/Ddl/TriggerFactory.php
+1. lib/internal/Magento/Framework/Data/FormFactory.php
+1. lib/internal/Magento/Framework/Data/Tree/NodeFactory.php
+1. lib/internal/Magento/Framework/Data/TreeFactory.php
+1. lib/internal/Magento/Framework/DataObject/Factory.php
+1. lib/internal/Magento/Framework/Event/ObserverFactory.php
+1. lib/internal/Magento/Framework/EventFactory.php
+1. lib/internal/Magento/Framework/File/UploaderFactory.php
+1. lib/internal/Magento/Framework/Filesystem/Directory/ReadFactory.php
+1. lib/internal/Magento/Framework/Mail/TransportInterfaceFactory.php
+1. lib/internal/Magento/Framework/Message/CollectionFactory.php
+1. lib/internal/Magento/Framework/Module/Setup/MigrationFactory.php
+1. lib/internal/Magento/Framework/Setup/Declaration/Schema/Diff/DiffFactory.php
+1. lib/internal/Magento/Framework/ValidatorFactory.php
+1. lib/internal/Magento/Framework/View/Asset/File/FallbackContextFactory.php
+1. lib/internal/Magento/Framework/View/Asset/FileFactory.php
+1. lib/internal/Magento/Framework/View/Layout/Generator/ContextFactory.php
+1. lib/internal/Magento/Framework/View/Layout/Reader/ContextFactory.php
+1. lib/internal/Magento/Framework/View/Page/ConfigFactory.php
+1. magento2ee/app/code/Magento/VisualMerchandiser/Model/Rules/Factory.php
+1. magento2ee/app/code/Magento/VisualMerchandiser/Model/Sorting/Factory.php
+
+
+# Blocks to be marked as API
+
+1. Magento\Catalog\Block\Navigation
```

Comment:
Most of this blocks are already marked as API (long time ago)

@paliarush can you please recheck the list

Reply:
@kandy could you please re-check the script used for collecting the list of blocks?